### ISSUE TITLE
requirement field for requestparam erroneously consider empty string as valid 

### ISSUE THREAD

**Issue by allan-simon** [FriendsOfSymfony/FOSRestBundle#991]
I got my action like this:

```php

    /*
     * @RequestParam(
     *     name="text",
     *     requirements="dummy",
     *     description="text of the reply (can't be empty)",
     *     strict=true,
     *     nullable=false
     * )
     * @View(statusCode=201) 
     *
     * @throws NotFoundHttpException|BadRequestHttpException
     */
    public function postVideoCommentReplyAction(
        ParamFetcherInterface $paramFetcher,
        $videoId,
        $commentId
    ) {
        $text = $paramFetcher->get('text');
   }
```

and when I perform a post request with 

```
{ 'text': '' } 
```

in the body , the text will be fetch to empty string , though  it should throw a bad request exception 

(of course if i put an actual value, the request fails  as expected,  same if put a bogus json that does not contain the text field ) 

I've made sure I got the last stable of the bundle (1.5.3)

**Comment by lsmith77** [comment]
can you submit a failing test case for this?

**Comment by allan-simon** [comment]
unfortunately I really don't think i will be able to give a nice failing test cases in the near future :/ 

however looking to the discussion #992  , it seems it's perfectly expected behaviour from symfony's developers point of view , so i don't know what to think in the case , is it a bug or a feature (i.e following what symfony's core does).

**Comment by lsmith77** [comment]
ping @webmozart

**Comment by Ener-Getick** [comment]
It seems to be the expected comportment. If you want to check that your param is not empty you must define ``allowBlank`` to ``false`` instead of ``nullable`` which just checks that your param is not strictly equals to ``null``.

```php
    /*
     * @RequestParam(
     *     name="text",
     *     requirements="dummy",
     *     description="text of the reply (can't be empty)",
     *     strict=true,
     *     allowBlank=false
     * )
     * @View(statusCode=201) 
     *
     * @throws NotFoundHttpException|BadRequestHttpException
     */
    public function postVideoCommentReplyAction(
        ParamFetcherInterface $paramFetcher,
        $videoId,
        $commentId
    ) {
        $text = $paramFetcher->get('text');
   }
```

**Comment by kopaygorodsky** [comment]
I also have a bug with an empty parameter in RequestParam. 
```
     /**
     * @param Team $team
     * @param $cardToken
     *
     * @return View
     * @Security("((user.hasTeamManageAccess(team) and is_granted('ROLE_USER')) or is_granted('ROLE_SUPER_ADMIN'))")
     * @RequestParam(name="cardToken", allowBlank=false, description="stripe card token")
     */
    public function patchCardAction(Team $team, $cardToken)
    {
    }
```

$cardToken is "", I have no idea why it's passing validation.

**Comment by Ener-Getick** [comment]
@kopaygorodsky bugs are no longer fixed in FOSRestBundle 1.8, you should upgrade to 2.0.
